compiler/src/dotty/tools/dotc/transform/patmat/Space.scala

```diff
@@ -937,12 +943,16 @@ object SpaceEngine {
                 && !pat.symbol.isAllOf(SyntheticCase, butNot=Method) // ExpandSAMs default cases use SyntheticCase
                 && isSubspace(covered, prev)
             then
-              val nullOnly = isNullable && rest.isEmpty && isWildcardArg(pat)
+              val nullOnly = isNullable && isWildcardArg(pat) && !mayCoverNull(prev)
               val msg = if nullOnly then MatchCaseOnlyNullWarning() else MatchCaseUnreachable()
               report.warning(msg, pat.srcPos)
 
             // in redundancy check, take guard as false in order to soundly approximate
-            val newPrev = if guard.isEmpty then covered :: prevs else prevs
+            val newPrev = if (guard.isEmpty)
```

Comment:
I'm wondering whether we can modify the `project` to avoid iterating over space. For wildcard pattern with nullable types, we project to a nullable space.

Reply:
I will try this approach, thanks.

Reply:
> I'm wondering whether we can modify the `project` to avoid iterating over space. For wildcard pattern with nullable types, we project to a nullable space.

Can you reword that, please?  I didn't understand either sentence.. 👉🏼 👈🏼

Reply:
I interpreted it as when the pattern is nullable, we project to a space with null included, like `targetSpace` on line 924.